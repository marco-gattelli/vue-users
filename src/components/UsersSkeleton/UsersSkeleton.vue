<template>
  <ul class="w-full" data-testid="users-skeleton">
    <li
      v-for="user in 4"
      :key="user"
      class="p-4 rounded border hover:bg-gray-50 transition-colors duration-500 my-4 w-full animate-pulse"
    >
      <div class="flex justify-between">
        <div class="flex flex-col gap-2">
          <span class="h-5 w-32 bg-gray-400 rounded"></span>
          <span class="h-5 w-44 bg-gray-400 rounded"></span>
          <span class="h-5 w-32 bg-gray-400 rounded"></span>
          <span class="h-5 w-44 bg-gray-400 rounded"></span>
        </div>
        <div class="flex flex-col gap-2 items-end">
          <span class="h-5 w-32 bg-gray-400 rounded"></span>
          <span class="h-5 w-44 bg-gray-400 rounded"></span>
        </div>
      </div>
    </li>
  </ul>
</template>
