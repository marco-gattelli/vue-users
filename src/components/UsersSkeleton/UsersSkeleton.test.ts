import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import UsersSkeleton from "@/components/UsersSkeleton/UsersSkeleton.vue";

describe("UsersSkeleton", () => {
  it("should render without crash", function () {
    // Arrange.
    // Act.
    const { element } = mount(UsersSkeleton);

    // Assert.
    expect(element).toMatchSnapshot();
  });
});
