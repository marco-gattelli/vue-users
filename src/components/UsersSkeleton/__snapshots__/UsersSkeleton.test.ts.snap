// Vitest Snapshot v1

exports[`UsersSkeleton > should render without crash 1`] = `
<ul
  class="w-full"
>
  
  <li
    class="p-4 rounded border hover:bg-gray-50 transition-colors duration-500 my-4 w-full animate-pulse"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
      </div>
      <div
        class="flex flex-col gap-2 items-end"
      >
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
      </div>
    </div>
  </li>
  <li
    class="p-4 rounded border hover:bg-gray-50 transition-colors duration-500 my-4 w-full animate-pulse"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
      </div>
      <div
        class="flex flex-col gap-2 items-end"
      >
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
      </div>
    </div>
  </li>
  <li
    class="p-4 rounded border hover:bg-gray-50 transition-colors duration-500 my-4 w-full animate-pulse"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
      </div>
      <div
        class="flex flex-col gap-2 items-end"
      >
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
      </div>
    </div>
  </li>
  <li
    class="p-4 rounded border hover:bg-gray-50 transition-colors duration-500 my-4 w-full animate-pulse"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
      </div>
      <div
        class="flex flex-col gap-2 items-end"
      >
        <span
          class="h-5 w-32 bg-gray-400 rounded"
        />
        <span
          class="h-5 w-44 bg-gray-400 rounded"
        />
      </div>
    </div>
  </li>
  
</ul>
`;
